### PR TITLE
automate GitHub releases

### PR DESCRIPTION
**Pull request by marco-neumann-by**
So nobody needs to do this manually.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -60,6 +60,7 @@ jobs:
 
     env:
       IS_MASTER_BUILD: ${{ !matrix.numfocus_nightly && matrix.os == 'ubuntu-latest' && matrix.pyarrow == '0.17.1' && matrix.python == '3.8' }}
+      IS_TAG: ${{ github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')}}
```
Comment (lr4d) [inline .github/workflows/ci.yml:63]: nitpick to make life simpler
```suggestion
      IS_TAG: ${{ github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')}}
      IS_RELEASE: ${{ env.IS_MASTER_BUILD == 'true' && env.IS_TAG == 'true }}
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -150,9 +151,28 @@ jobs:
           # NOTE: `token` is not required, because the kartothek repo is public
           file: ./coverage.xml
           name: pytest-numfocus_nightly_${{ matrix.numfocus_nightly }}-os_${{ matrix.os }}-pyarrow_${{ matrix.pyarrow }}-python_${{ matrix.python }}
+
+      # Release
       - name: Publish to PyPI
-        if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags') && env.IS_MASTER_BUILD == 'true'
+        if: env.IS_MASTER_BUILD == 'true' && env.IS_TAG == 'true'
```
Comment (lr4d) [inline .github/workflows/ci.yml:157]:
```suggestion
        if: env.IS_RELEASE == 'true'
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -150,9 +151,28 @@ jobs:
...
         uses: pypa/gh-action-pypi-publish@master
         with:
           user: __token__
           password: ${{ secrets.PYPI_PASSWORD }}
+      - name: Create Release Notes
+        if: env.IS_MASTER_BUILD == 'true' && env.IS_TAG == 'true'
```
Comment (lr4d) [inline .github/workflows/ci.yml:163]:
```suggestion
        if: env.IS_RELEASE == 'true'
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -150,9 +151,28 @@ jobs:
...
+        run: |
+            conda install pandoc
+            awk 'BEGIN{found=0} {if (match($0, "==============")) {if (found == 1) exit; found=1}; if (found == 1) {print last}; last=$0}' CHANGES.rst > release_notes.rst
+            pandoc --from=rst --to=markdown -O release_notes.md release_notes.rst
+      - name: Create GitHub Release
+        if: env.IS_MASTER_BUILD == 'true' && env.IS_TAG == 'true'
```
Comment (lr4d) [inline .github/workflows/ci.yml:169]:
```suggestion
        if: env.IS_RELEASE == 'true'
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (marco-neumann-by) [reply .github/workflows/ci.yml:63]: This doesn't work. You canny use `env` in the `env` section. I've tried that before :D

File: .github/workflows/ci.yml (hunk 3 above)
Comment (marco-neumann-by) [reply .github/workflows/ci.yml:163]: see above

File: .github/workflows/ci.yml (hunk 2 above)
Comment (marco-neumann-by) [reply .github/workflows/ci.yml:157]: see above

File: .github/workflows/ci.yml (hunk 4 above)
Comment (marco-neumann-by) [reply .github/workflows/ci.yml:169]: see above

**Pull request merged**